src/util/triplea/MapXMLCreator/MapXMLHelper.java

```diff
@@ -0,0 +1,1008 @@
+package util.triplea.MapXMLCreator;
+
+import games.strategy.engine.data.GameParser;
+import games.strategy.util.Triple;
+import games.strategy.util.Tuple;
+
+import java.awt.HeadlessException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map.Entry;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import util.image.FileOpen;
+import util.image.FileSave;
+import util.triplea.MapXMLCreator.TerritoryDefinitionDialog.DEFINITION;
+
+import com.sun.org.apache.xerces.internal.impl.xpath.regex.ParseException;
+
+/**
+ * This class reads, writes and keeps the Map XML properties.
+ * 
+ * @author Erik von der Osten
+ * 
+ */
+public class MapXMLHelper {
+
+	private static final String TRIPLEA_JAVA_CLASS_DELEGATE_PATH = "games.strategy.triplea.delegate.";
+	static LinkedHashMap<String, String> s_xmlStrings = new LinkedHashMap<String, String>();
+	static ArrayList<String> s_resourceList = new ArrayList<String>();
+	static HashMap<String, HashMap<DEFINITION, Boolean>> s_territoyDefintions = new HashMap<String, HashMap<DEFINITION, Boolean>>();

```

Comment:
Do not code against specific implementations unless you need to, code against an API, not implementations. IN this case that means the left hand side should only be as specific as it needs to be. Unless you need methods specifically from "HashMap", use 'Map' instead. Same thing for "ArrayList" and "list" and "linkedHashMap" and "Map"